batch/batch/worker/worker.py

```diff
@@ -2627,14 +2627,28 @@ async def execute(self, classpath: str, scratch_dir: str, log_file: str, jar_url
             elif message == JVM.FINISH_USER_EXCEPTION:
                 exception = await read_str(reader)
                 raise JVMUserError(exception)
-            elif message == JVM.FINISH_ENTRYWAY_EXCEPTION:
-                log.warning(f'{self}: entryway exception encountered (interrupted: {wait_for_interrupt.done()})')
-                exception = await read_str(reader)
-                raise ValueError(exception)
-            elif message == JVM.FINISH_JVM_EOS:
-                assert eos_exception is not None
-                log.warning(f'{self}: unexpected end of stream in jvm (interrupted: {wait_for_interrupt.done()})')
-                raise ValueError('unexpected end of stream in jvm') from eos_exception
+            else:
+                jvm_output = ''
+                if os.path.exists(self.container.container.log_path):
+                    jvm_output = (await self.fs.read(self.container.container.log_path)).decode('utf-8')
+
+                if message == JVM.FINISH_ENTRYWAY_EXCEPTION:
+                    log.warning(
+                        f'{self}: entryway exception encountered (interrupted: {wait_for_interrupt.done()})\nJVM Output:\n\n{jvm_output}'
+                    )
+                    exception = await read_str(reader)
+                    raise ValueError(exception)
+                if message == JVM.FINISH_JVM_EOS:
+                    assert eos_exception is not None
+                    log.warning(
+                        f'{self}: unexpected end of stream in jvm (interrupted: {wait_for_interrupt.done()})\nJVM Output:\n\n{jvm_output}'
+                    )
+                    raise ValueError(
+                        # Do not include the JVM log in the exception as this is sent to the user and
+                        # the JVM log might inadvetantly contain sensitive information.
+                        'unexpected end of stream in jvm'
+                    ) from eos_exception
+                assert False, f'unexpected message type: {message}\nJVM Output:\n\n{jvm_output}'
```

Comment:
I feel like we can do something more than `assert False`. Can we instead `log.exception` and raise some `ValueError` akin to our `HailInternalError` that tells a user they should reach out to us.

Reply:
I added a log and switched from AssertionError to ValueError. Do you want everything switched to inform the user to contact us? All three of these branches (entryway exception, JVM EOS, and invalid message) are generally not user caused (except maybe the JVM EOS if the user has large data rows).